satpy/tests/enhancement_tests/test_enhancements.py

```diff
@@ -501,75 +501,145 @@ def fake_area():
+            'ci_prob30', 'ci_prob60', 'ci_prob90', 'asii_turb_trop_prob',
+            'MapCellCatType']
         )
-def test_producing_mode_p(fake_area, tmp_path, data):
-    """Test producing mode p with  palettizer and ancillary variables."""
+def test_nwcsaf_comps(fake_area, tmp_path, data):
+    """Test loading NWCSAF composites."""
     from satpy.writers import get_enhanced_image
 
     from ... import Scene
-    (palette, comp, label, dtp) = _nwcsaf_props[data]
+    (flavour, dvname, altname, palettename, statusname, comp, filelabel, dtp) = _nwcsaf_geo_props[data]
     rng = (0, 100) if dtp == "uint8" else (-100, 1000)
-    fk = tmp_path / f"S_NWC_{label:s}_MSG2_MSG-N-VISIR_20220124T094500Z.nc"
+    if flavour == "geo":
+        fn = f"S_NWC_{filelabel:s}_MSG2_MSG-N-VISIR_20220124T094500Z.nc"
+        reader = "nwcsaf-geo"
+        id_ = {"satellite_identifier": "MSG4"}
+    else:
+        fn = f"S_NWC_{filelabel:s}_noaa20_00000_20230301T1200213Z_20230301T1201458Z.nc"
+        reader = "nwcsaf-pps_nc"
+        id_ = {"platform": "NOAA-20"}
+    fk = tmp_path / fn
     # create a minimally fake netCDF file, otherwise satpy won't load the
     # composite
     ds = xr.Dataset(
             coords={"nx": [0], "ny": [0]},
             attrs={
                 "source": "satpy unit test",
-                "satellite_identifier": "pranksat",
                 "time_coverage_start": "0001-01-01T00:00:00Z",
-                "time_coverage_end": "0001-01-01T01:00:00Z"
-                })
+                "time_coverage_end": "0001-01-01T01:00:00Z",
+                } | id_,
+            )
     ds.to_netcdf(fk)
-    sc = Scene(filenames=[os.fspath(fk)], reader=["nwcsaf-geo"])
-    sc[palette] = xr.DataArray(
+    sc = Scene(filenames=[os.fspath(fk)], reader=[reader])
+    sc[palettename] = xr.DataArray(
             da.tile(da.arange(256), [3, 1]).T,
             dims=("pal02_colors", "pal_RGB"))
     fake_alti = da.linspace(rng[0], rng[1], 4, chunks=2, dtype=dtp).reshape(2, 2)
-    sc[data] = xr.DataArray(
+    ancvars = [sc[palettename]]
+    if statusname is not None:
+        sc[statusname] = xr.DataArray(
+                da.zeros(shape=(2, 2), dtype="uint8"),
+                attrs={
+                    "area": fake_area,
+                    "_FillValue": 123},
+                dims=("y", "x"))
+        ancvars.append(sc[statusname])
+    sc[dvname] = xr.DataArray(
             fake_alti,
             dims=("y", "x"),
             attrs={
                 "area": fake_area,
-                "ancillary_variables": [sc[palette]],
+                "ancillary_variables": ancvars,
                 "valid_range": rng})
-    sc.load([comp])
+
+    def _fake_get_varname(info, info_type="file_key"):
+        return altname or dvname
+    with mock.patch("satpy.readers.nwcsaf_nc.NcNWCSAF._get_varname_in_file") as srnN_:
+        srnN_.side_effect = _fake_get_varname
+        sc.load([comp])
```

Comment:
This mock is not pretty, but otherwise the reader is actually looking into the NetCDF file, and I wanted to avoid creating a fully realistic fake NetCDF file.  Trying to write tests without testing on real data files is already creating enough work as it is.